Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/FollowMeToggle.cs

```diff
@@ -1,40 +1,183 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using Microsoft.MixedReality.Toolkit.Utilities;
 using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.UI
 {
     [RequireComponent(typeof(RadialView))]
     public class FollowMeToggle : MonoBehaviour
     {
+        /// <summary>
+        /// An optional object for visualizing the 'Follow Me' mode state.
+        /// </summary>
+        public GameObject VisualizationObject
+        {
+            get { return visualizationObject; }
+            set { visualizationObject = value; }
+        }
+
         [SerializeField]
-        [Tooltip("An opional object for visualizing the 'Follow Me' mode state")]
+        [Tooltip("An optional object for visualizing the 'Follow Me' mode state.")]
         private GameObject visualizationObject = null;
 
+        /// <summary>
+        /// An optional Interactable to select/deselect when toggling the follow behavior.
+        /// </summary>
+        public Interactable InteractableObject
+        {
+            get { return interactableObject; }
+            set { interactableObject = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("An optional Interactable to select/deselect when toggling the follow behavior.")]
+        private Interactable interactableObject = null;
+
+        /// <summary>
+        /// Should following be automatically enabled when the user is further than a certain distance away?
+        /// </summary>
+        public bool AutoFollowAtDistance
+        {
+            get { return autoFollowAtDistance; }
+            set
+            {
+                if (autoFollowAtDistance != value)
+                {
+                    autoFollowAtDistance = value;
+
+                    if (autoFollowAtDistance)
+                    {
+                        autoFollowDistanceCheck = StartCoroutine(AutoFollowDistanceCheck());
```

Comment:
Do you need to stop the previous coroutine before starting this one?

Reply:
I don't believe stop needs to be called due to the above `if (autoFollowAtDistance != value) ` conditional safeguarding that case.